internal/controller/pullrequest_controller.go

```diff
@@ -58,121 +58,92 @@ type PullRequestReconciler struct {
 // - https://pkg.go.dev/sigs.k8s.io/controller-runtime@v0.17.2/pkg/reconcile
 func (r *PullRequestReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Result, error) {
 	logger := log.FromContext(ctx)
+	logger.Info("Reconciling PullRequest")
 
 	var pr promoterv1alpha1.PullRequest
-	err := r.Get(ctx, req.NamespacedName, &pr, &client.GetOptions{})
-	if err != nil {
+	if err := r.Get(ctx, req.NamespacedName, &pr); err != nil {
 		if errors.IsNotFound(err) {
 			logger.Info("PullRequest not found", "namespace", req.Namespace, "name", req.Name)
 			return ctrl.Result{}, nil
 		}
-
-		logger.Error(err, "failed to get PullRequest", "namespace", req.Namespace, "name", req.Name)
 		return ctrl.Result{}, fmt.Errorf("failed to get PullRequest: %w", err)
 	}
 
-	pullRequestProvider, err := r.getPullRequestProvider(ctx, pr)
+	provider, err := r.getPullRequestProvider(ctx, pr)
 	if err != nil {
 		return ctrl.Result{}, fmt.Errorf("failed to get PullRequest provider: %w", err)
 	}
-	if pullRequestProvider == nil {
-		return ctrl.Result{}, fmt.Errorf("failed to get pull request provider, pullRequestProvider is nil")
-	}
 
-	deleted, err := r.handleFinalizer(ctx, &pr, pullRequestProvider)
-	if err != nil {
-		return ctrl.Result{}, fmt.Errorf("failed to handle finalizer: %w", err)
-	}
-	if deleted {
-		return ctrl.Result{}, nil
+	if deleted, err := r.handleFinalizer(ctx, &pr, provider); err != nil || deleted {
+		return ctrl.Result{}, err
 	}
 
-	found, err := pullRequestProvider.FindOpen(ctx, &pr)
+	logger.Info("Checking for open PR on provider")
+	found, err := provider.FindOpen(ctx, &pr)
 	if err != nil {
 		return ctrl.Result{}, fmt.Errorf("failed to check for open PR: %w", err)
 	}
 
-	// We can't find an open PR on the provider, and we have had a status state before, we should now delete it because
-	// it no longer exists on provider.
 	if !found && pr.Status.State != "" {
-		logger.Info("Deleting Pull Request, because no open PR found on provider")
-		err = r.Delete(ctx, &pr)
-		if err != nil {
-			if errors.IsNotFound(err) {
-				return ctrl.Result{}, nil
-			}
+		logger.Info("Deleting Pull Request, no open PR found on provider")
+		if err := r.Delete(ctx, &pr); err != nil && !errors.IsNotFound(err) {
+			logger.Error(err, "Failed to delete PullRequest")
 			return ctrl.Result{}, fmt.Errorf("failed to delete PullRequest: %w", err)
 		}
 		return ctrl.Result{}, nil
 	}
 
-	// If we have a status state, and the status state is the same as the spec state, and the observed generation is the same as the generation, we don't need to reconcile.
-	if pr.Status.State != "" && pr.Spec.State == pr.Status.State && pr.Status.ObservedGeneration == pr.Generation {
+	if pr.Status.State == pr.Spec.State && pr.Status.ObservedGeneration == pr.Generation {
```

Comment:
gitRepositoryRef, title, and description are all mutable... do we need to confirm that status matches spec for those as well?

Reply:
Actually so the generation check resolves title/description.

Reply:
```suggestion
	// If there hasn't been an SCM change to PR state, and there also hasn't been a spec change,
	// quit early. It's only possible to hit this condition if the PR is open on the SCM. Otherwise the
	// CR would have been deleted and reconciliation short-circuited.
	if pr.Status.State == pr.Spec.State && pr.Status.ObservedGeneration == pr.Generation {
```